database/src/ops/tx.rs

```diff
@@ -128,7 +128,7 @@ pub fn add_tx(
     };
 
     let mut amount_indices = Vec::with_capacity(tx.prefix.outputs.len());
-    let tx_idx = get_num_tx(tables.tx_ids_mut())?;
+    let tx_idx = get_num_tx(tables.tx_ids())?.saturating_sub(1);
```

Comment:
I think this change is correct? It was causing problems in tests (1st transaction added had index of 1).

Reply:
I don't think so, this is shadowing the correct index at the top of this function, was that intentional?